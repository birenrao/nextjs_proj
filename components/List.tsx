import * as React from 'react'
import ListItem from './ListItem'
import { Movies } from '../interfaces';
import '../styles/list.scss';

type Props = {
  items: Movies[];
};

const List: React.FunctionComponent<Props> = ({ items }) => (
  <div className="grid container">
    {items && items.map(item => (
      <div className="item" key={item.id}>
        <ListItem data={item} />
      </div >
    )) || <div className="errorMSg">Error: due to some technical issues movies details are not available  </div> }
  </div>
)

export default List;
