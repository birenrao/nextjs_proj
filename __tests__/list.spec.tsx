import * as React from 'react';
import { mount } from 'enzyme';
import List from '../components/List';
import results from './items';

let tree: any;
const result: any = results && results || [];
beforeEach(() => {
  tree = mount(<List items={result} />);
});
describe('components', () => {
  describe('List', () => {
    it('renders List component correctly', () => {
      expect(tree.debug()).toMatchSnapshot();
    });
  });
});
