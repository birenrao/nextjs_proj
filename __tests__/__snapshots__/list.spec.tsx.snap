// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components List renders List component correctly 1`] = `
"<List items={{...}}>
  <div className=\\"grid container\\" />
</List>"
`;
